documentation/user-consent.md

```diff
@@ -0,0 +1,220 @@
+-	`Legal Entities`: are the legal subjects that are willing to get access to personal information with a specific, predefined purpose.
+
+>[TO BE EDITED/COMPLETED]
+
+## Purposes
+
+A purpose declares what the application intends to do with a set of Personal Information resources. It provides a human-friendly description of why the data processing takes place and a list of the scopes of that Personal Information that gives access to its processing.
+
+Purposes can be grouped into three categories: 
+
+- **Automatic purpose** : This purpose does not require explicit consent from the user and cannot be revoked. For example, you need to handle call logs to create an invoice.
+- **Opt-out purpose** : This purpose does not require explicit consent from the user, but consent can be revoked. For example, processing invoices to improve our commercial offering and for marketing purposes.
+- **Opt-in purpose** : This purpose requires explicit user consent and can be revoked. For example, creating a credit score for a user based on CDRs and payment history.
+
+Moreover, when it comes to granting consent, there are two types of the purposes:
+
+-	**User-bound**: the consent is granted to be applied for all the identifiers of a given user i.e. granted at user level. For example, grant consent for accessing to the call history of any of her phone numbers. 
+-	**Identifier-bound**: the consent is granted to be applied only for a specific user’s identifier i.e. for a public identifier associated with the service the Telco Operator provides to the user (typically a phone number). For example, grant consent for accessing to the call history of only one of her phone numbers.
+
+>[TO BE EDITED/COMPLETED]
+
+### Purpose levels
+
+Depending on the purpose categories the purpose may have a specific level that determines the processing allowed by the user if the purpose is granted. **User consent is required to gain access to non-automatic purposes**.
+
+| **Purpose Level** | Description | Category |
+| --- | --- | --- |
+|`legal_obligation` | A process that is required by law. | Automatic purpose |
+|`contract` | A process that is needed to fulfill the service contract with the user, and it is already approved when the contract was signed. | Automatic purpose |
+|`legitimate_interest`| An interest of the data processor that does not conflict with the fundamental interest of the individuals. | Opt-out purpose |
+|`compatible` | A re-processing of data acquired through another purpose that is compatible with the first purpose. | Opt-out purpose|
+|`consent` | This level applies to purposes which are unrelated enough to the service. | Opt-in purpose|
+|`terms_and_conditions`| The operations or tasks done by an application to provide its functionality, as described in the Terms and Conditions. | Opt-in purpose|
```

Comment:
Are there any reference for proposed **Purpose levels**?
If the levels are ordered from the most strict to less significant, then the `consent` level should be the last one, I guess.

The description for `compatible` level is not clear - what is *another purpose* and what is *first purpose*.
The name for `consent` level is ambiguous since **User consent is required to gain access to all non-automatic purposes**. Can it be called `voluntary` or `optional` or other?

Reply:
Will include this changes if it is ok with you @rartych

Reply:
Fixed in https://github.com/camaraproject/IdentityAndConsentManagement/pull/13/commits/4b761353f33ee263e98e8309c8fb698908401206